src/main/java/edu/ie3/datamodel/models/input/container/GraphicElements.java

```diff
@@ -52,6 +53,44 @@ public GraphicElements(Collection<GraphicElements> graphicElements) {
     }
   }
 
+  /**
+   * Create an instance based on a list of {@link GraphicInput} entities that are included in {@link
+   * GraphicElements}
+   *
+   * @param graphics list of grid elements this container instance should created from
+   */
+  public GraphicElements(List<GraphicInput> graphics) {
+
+    /* init sets */
+    this.nodeGraphics = new HashSet<>();
+    this.lineGraphics = new HashSet<>();
+
+    /* fill the sets */
+    graphics.forEach(
+        systemParticipantInput -> {
+          if (systemParticipantInput instanceof NodeGraphicInput) {
+            nodeGraphics.add((NodeGraphicInput) systemParticipantInput);
+          } else if (systemParticipantInput instanceof LineGraphicInput) {
+            lineGraphics.add((LineGraphicInput) systemParticipantInput);
+          } else {
+            throw new InvalidEntityException(
+                "Provided entity is not included in GraphicElements.", systemParticipantInput);
+          }
+        });
+
+    // sanity check for distinct uuids
+    Optional<String> exceptionString =
+        ValidationUtils.checkForDuplicateUuids(new HashSet<>(this.allEntitiesAsList()));
+    if (exceptionString.isPresent()) {
+      throw new InvalidGridException(
+          "The provided entities in '"
+              + this.getClass().getSimpleName()
+              + "' contains duplicate UUIDs. "
+              + "This is not allowed!\nDuplicated uuids:\n\n"
+              + exceptionString);
+    }
+  }
+
   @Override
   public List<GraphicInput> allEntitiesAsList() {
```

Comment:
```suggestion
  public final List<GraphicInput> allEntitiesAsList() {
```

Might resolve the SonarQube issue!?